[Example] Fix error in gcn train due to missing argument

europeanplaice opened the pull request:
## Description
Fixes #3039

I finished coding on this PR.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] Related issue is referred in this PR

## Changes
* Add `parser.add_argument` to receive a dataset name. (default `cora`)
* When name `dataset` was used as an argument, error `argparse.ArgumentError: argument --dataset: conflicting option string: --dataset` was raised. Therefore I changed `dataset` to `dataset_name`.
* Remove unused imports

jermainewang commented:
Hi @europeanplaice , thanks for the fix! The issue you raised makes a lot of sense. We shouldn't let the script fail with an awkward (and confusing) `None` dataset. Here are two of my suggestions:

* We are in the process of deprecating the usage of `register_data_args(parser)`. Since you are modifying these examples anyway, could you help delete the corresponding lines? This is also the source of the duplicate argument error you've encountered, so you should be able to use the `--dataset` argument once again after this change.
* In the argument description, it is better to also give the allowed choices. For example `"Dataset name ('cora', 'citeseer', 'pubmed')."`.

europeanplaice commented:
Thank you for the review!

I had missed `register_data_args` and now have found how this function works.
I removed this function and reverted the argument `dataset_name` to `dataset`.
I also modified the argument description as you suggested.

I have chosen `cora` as the default dataset, however, I'm not sure if this is preferable.
Otherwise, should I remove `default` and set `required`?

jermainewang commented:
> Otherwise, should I remove `default` and set `required`?

I think either is ok. Thanks again for the fix!
